sw/airborne/modules/guidance/gvf/gvf.c

```diff
@@ -237,6 +238,9 @@ bool gvf_ellipse(uint8_t wp, float a, float b, float alpha)
     gvf_trajectory.p[3] = 60;
   }
 
+  if (gvf_trajectory.p[2] == gvf_trajectory.p[3])
+    horizontal_mode = HORIZONTAL_MODE_CIRCLE;
```

Comment:
this mode should be set to HORIZONTAL_MODE_WAYPOINT if not a circle nor a segment